PurchasesTests/Purchasing/ProductInfoTests.swift

```diff
@@ -0,0 +1,273 @@
+import XCTest
+import Nimble
+
+import Purchases
+
+class ProductInfoTests: XCTestCase {
+    func testAsDictionaryConvertsProductIdentifierCorrectly() {
+        let product_identifier = "cool_product"
+        let productInfo = RCProductInfo(productIdentifier: product_identifier,
+                                        paymentMode: .none,
+                                        currencyCode: "UYU",
+                                        price: 9.99,
+                                        normalDuration: nil,
+                                        introDuration: nil,
+                                        introDurationType: .none,
+                                        introPrice: nil,
+                                        subscriptionGroup: nil,
+                                        discounts: nil)
```

Comment:
could you make a function that builds an empty `RCProductInfo` and then modify each field that you're testing? Something like:

```
let productInfo = emptyProductInfo()
productInfo.productIdentifier = "cool_product"
```

Reply:
related to https://github.com/RevenueCat/purchases-ios/pull/256#discussion_r436099550, 
I'll do the refactor, but it can't really contain an empty product info because of the reasons mentioned there

Reply:
Yeah, I was actually referring to create a function that builds:

```RCProductInfo(productIdentifier: "cool_product",
                                        paymentMode: .none,
                                        currencyCode: "UYU",
                                        price: price,
                                        normalDuration: nil,
                                        introDuration: nil,
                                        introDurationType: .none,
                                        introPrice: nil,
                                        subscriptionGroup: nil,
                                        discounts: nil)
```

And then modify each property. It's not an `emptyProductInfo`, it's more a `validProductInfo`, as you mentioned 👍

Reply:
done!